### PR TITLE
601 Swagger 2 PoC

### PR DESCRIPTION
**Pull request by Zejnilovic**
For other projects I PoCed Swagger. Used Enceladus as a PoC Base. Leaving this base here. Anybody might pick it up, even me. Need just more configuration to what API it should map and of course properly documented APIs.

Just build, run and go to `http://localhost:8080/menas/swagger-ui.html` or `IP/where-is-the-war/swagger-ui.html`. You need to log in to Menas first.

Screenshot 2021-03-04 at 14 41 30

### CODE REVIEW

File: menas/src/main/scala/za/co/absa/enceladus/menas/SpringFoxConfig.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  def api(): Docket = {
+    new Docket(DocumentationType.SWAGGER_2)
+      .select
+      .apis(RequestHandlerSelectors.any)
+      .paths(filteredPaths)
+      .build
+  }
+
+  private def filteredPaths: Predicate[String] =
+    or[String](regex("/api/dataset.*"), regex("/api/schema.*"))
```
Comment (lokm01) [inline menas/src/main/scala/za/co/absa/enceladus/menas/SpringFoxConfig.scala:37]: Not sure what this does, but mapping table?

**Review by lokm01 — COMMENTED** [review]
Love the idea of having this!

File: menas/src/main/scala/za/co/absa/enceladus/menas/SpringFoxConfig.scala (hunk 1 above)
Comment (Zejnilovic) [reply menas/src/main/scala/za/co/absa/enceladus/menas/SpringFoxConfig.scala:37]: and properties, and login, and others

File: menas/src/main/scala/za/co/absa/enceladus/menas/SpringFoxConfig.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  private def apiInfo =
+    new ApiInfoBuilder()
+      .title("Menas API")
+      .description("Menas API reference for developers")
+      .license("Apache 2.0 License")
+      .version("1.0") // api or project?
```
Comment (Zejnilovic) [inline menas/src/main/scala/za/co/absa/enceladus/menas/SpringFoxConfig.scala:36]: Has much more methods to add metadata

**Comment by benedeki** [comment]
I consider this really cool 😎

**Review by AdrianOlosutean — COMMENTED** [review]
It's quite nice

**Comment by Zejnilovic** [comment]
Implemented in #1739

**Pull request closed without merging**